### PR TITLE
ORCID harvester.

### PR DESCRIPTION
**Pull request by justinlittman**
refs #1268

## Why was this change made?
To allow creating publications from a researcher's ORCID profile.

## How was this change tested?
Unit

## Which documentation and/or configurations were updated?
NA

### CODE REVIEW

**Review by jmartin-sul — COMMENTED** [review]
two questions (either/both of which might be me missing something), one nitpick, overall LGTM

File: lib/orcid/harvester.rb [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    def process_works(author, groups_response)
+      groups_response.map { |work_response| process_work(author, work_response) }.compact
+    end
+
+    def process_work(author, work_summary_response)
+      # Using first work summary only.
+      work_summary = WorkRecord.new(work_summary_response['work-summary'][0])
+
+      return unless check_work_type?(author, work_summary) && check_external_ids?(author, work_summary)
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply lib/orcid/harvester.rb:74]: The API call that we make will only return works where the visibility is "public," so no need to check here.

File: lib/orcid/harvester.rb [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        pub_hash: WorkMapper.map(work),
+        orcid_source_record: source
+      )
+    end
+
+    def find_matching_publication(work)
+      work.external_ids.map do |external_id|
+        PublicationIdentifier.find_by(identifier_type: IdentifierTypeMapper.to_sul_pub_id_type(external_id.type),
+                                      identifier_value: external_id.value)&.publication
+      end.compact.first
```
[the remark this answers is not in the document]
Comment (justinlittman) [reply lib/orcid/harvester.rb:122]: It is missing; I changed my mind and decided it wasn't necessary. (I had originally thought this was an identifier for the publication; however, it is really an identifier of a publication for a researcher.)

File: lib/orcid/harvester.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply lib/orcid/harvester.rb:74]: ah, got it, thanks

File: lib/orcid/harvester.rb [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+    def new_publication(work, source)
+      Publication.new(
+        active: true,
+        pub_hash: WorkMapper.map(work),
+        orcid_source_record: source
+      )
+    end
+
+    def find_matching_publication(work)
```
[the remark this answers is not in the document]
Comment (jmartin-sul) [reply lib/orcid/harvester.rb]: thanks!

**Review by jmartin-sul — APPROVED** [review]

**Pull request merged**